rpc/server.go

```diff
@@ -93,6 +100,21 @@ func (s *Server) ServeCodec(codec ServerCodec, options CodecOption) {
 // is used to serve HTTP connections. Subscriptions and reverse calls are not allowed in
 // this mode.
 func (s *Server) serveSingleRequest(ctx context.Context, codec ServerCodec) {
+
+	if s.maxConReq > 0 {
+		if s.reqCount > s.maxConReq { //if maxConReq is 0, then no limit
+			maxConcReqDiscardedTxs.Inc(1)
+			log.Warn("RPC server ratelimiting", "Concurrent Reqs", s.reqCount, "MaxConcurrentReqs", s.maxConReq)
```

Comment:
I believe we should use atomic here as well. To prevent such things, could you use https://pkg.go.dev/sync/atomic#Int64 instead of plain int64?

Reply:
Done !